File: lesson_8/app/src/main/java/com/kalashnyk/denys/gmapsapp/di/component/ApiComponent.java
package com.kalashnyk.denys.gmapsapp.di.component;

import com.kalashnyk.denys.gmapsapp.di.module.ApiModule;
import com.kalashnyk.denys.gmapsapp.di.scope.ApiScope;
import com.kalashnyk.denys.gmapsapp.repository.server.ServerCommunicator;

import dagger.Component;

@ApiScope
@Component(modules = {ApiModule.class})
public interface ApiComponent {
    ServerCommunicator getServerCommunicator();
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/di/component/ApiComponent.kt
package com.kalashnyk.denys.kotlinbindingsample.di.component

import com.kalashnyk.denys.kotlinbindingsample.di.module.ApiModule
import com.kalashnyk.denys.kotlinbindingsample.di.scope.ApiScope
import com.kalashnyk.denys.kotlinbindingsample.repository.server.ServerCommunicator
import dagger.Component

@ApiScope
@Component(modules = [ApiModule::class])
interface ApiComponent {
    val serverCommunicator: ServerCommunicator
}
